### DISCUSSION TITLE
HELP with importing DTED in earth file for AFSIM

### DISCUSSION THREAD

**Discussion by nixongr14** [pelicanmapping/osgearth#2436] [Support Q&A] [answered]
Hello all! 

So I'm working with a modeling and simulation tool called AFSIM. AFSIM can read in .earth files and update a map users can see during playbacks. Now I'd like to see my dted/terrain heights in also in my replays. 

Now one doesn't need to know much about AFSIM, expect that AFSIM currently excepts osgEarth version 2.7. Based on my research here is my earth file. 

<map type="geocentric" version="2">   
    <options>
        <terrain vertical_scale="3"/>
    </options>
     
    <image name="test" driver="gdal">
        <url>my_world.tif</url>
    </image>

    <heightfield name="terrain" driver="gdal">
        <url>data\all_terrain</url>
    </heightfield>   
</map> 

Can someone help me? What's the right way to make this file to visualize the dted and terrain height.

**Comment by gwaldron** [comment, marked as answer]
That's a pretty OLD version so memory is hazy. But two things:

DTED data is expressed relative to the EGM96 geoid, so you need to tell osgEarth as much:
```xml
<heightfield name="terrain" driver="gdal">
    <vdatum>egm96</vdatum>
   ...
```
Old versions of osgEarth used to support the loading of an entire directory of files, but we removed that because it was very problematic. The correct (and forward-compatible) approach instead is the create a GDAL VRT (using the [gdalbuildvrt](https://gdal.org/programs/gdalbuildvrt.html) tool) and reference that in your `<url>` property.
```xml
   <url>myterrain.vrt</url>
```

Obviously I would recommend upgrading to a newer osgEarth if at all possible. [There's a helpful guide here](https://docs.osgearth.org/en/latest/3.0_upgrade_guide.html).

Good luck.

**Reply by nixongr14** [reply]
Thank you Glenn it worked! 

Sadly AFSIM is managed by the Air Force so I don't see any update coming soon.

**Reply by XenonofArcticus** [reply]
At one time AlphaPixel was approved to work on AFSIM, if you ever want anything done to it.
